app/models/user_group.rb

```diff
@@ -180,6 +180,16 @@ def is_root_group?
     parent_group_id.nil?
   end
 
+  def self.roles_of_group_type(group_type, current_user: nil, params: {}, includes_params: [])
+    roles = UserRole.includes(includes_params + [:group]).where(group: { group_type: group_type })
```

Comment:
Nitpick: I think `UserRole.includes(:group).includes(includes_params)` is a bit more straight forward than manually glueing arrays together (especially since you cannot be sure that `includes_params` itself will always be an array. Passing hashes as include_params is totally valid in Rails.)